lib/matplotlib/tests/test_path.py

```diff
@@ -261,14 +267,33 @@ def test_xkcd_marker():
     y1 = x
     y2 = 5 - x
     y3 = 2.5 * np.ones(8)
-
     with plt.xkcd():
         fig, ax = plt.subplots()
         ax.plot(x, y1, '+', ms=10)
         ax.plot(x, y2, 'o', ms=10)
         ax.plot(x, y3, '^', ms=10)
 
 
+@image_comparison(['xkcd_artist.png'], remove_text=True)
+def test_xkcd_artist():
+    with plt.xkcd():
+        rcParams['path.sketch'] = (2, 120, 40)
+        artist.Artist().set_sketch_params(seed=19680801)
+        fig, ax = plt.subplots()
+        ax.pie([10, 20, 30, 40], wedgeprops={'edgecolor': 'black'})
+        ax.plot()
+
+
+@image_comparison(['xkcd_seed.png'], remove_text=True)
+def test_xkcd_seed():
+    with plt.xkcd():
+        rcParams['path.sketch'] = (2, 120, 40)
+        rcParams['path.sketch_seed'] = 2000
+        fig, ax = plt.subplots()
+        ax.pie([10, 20, 30, 40], wedgeprops={'edgecolor': 'black'})
+        ax.plot()
+
```

Comment:
Have added the test:
```
x = np.linspace(0.7,1.42,100)
    y_1 = [0.7]*100
    y_2 = 10*x**0.2-x**1.5
    y_3 = 20*np.sin(x)
    fig, ax = plt.subplots()
    ln = ax.plot(x, y_1, color = 'blue')
    ln[0].set_sketch_params(3, 120, 40, 2000)

    with plt.xkcd(seed=19680801):
        ln_2 = ax.plot(x, y_2, color = 'green')
        ln_2[0].set_sketch_params(3,120,40)
    
    rcParams['path.sketch_seed'] = 645
    ln_3 = ax.plot(x, y_3, color = 'red')
    ln_3[0].set_sketch_params(3,120,40)
```
Each part tests each way of setting the seed.
The baseline image is:
![xkcd_seed](https://github.com/matplotlib/matplotlib/assets/77437286/4ef29121-222a-4e4a-91c0-3ad656da0e2d)
3 lines with 3 seeds set in 3 different ways.

Reply:
I've split the tests, now the tests also test the interaction of one seed updation method with the other.

Reply:
For reference, these are the tests that were added 

```
@image_comparison(['artist_seed.png'], remove_text=True)
def test_artist_seed():
    x = np.linspace(0.7, 1.42, 100)
    y_1 = [0.7]*100
    y_2 = 10*x**0.2-x**1.5
    y_3 = 20*np.sin(x)
    fig, ax = plt.subplots()
    ln = ax.plot(x, y_1, color='blue')
    ln[0].set_sketch_params(3, 120, 40, 2000)

    # set_sketch_params seed should override seed set by rcParam
    rcParams['path.sketch_seed'] = 0
    ln_2 = ax.plot(x, y_2, color='green')
    ln_2[0].set_sketch_params(3, 120, 40, 19680801)

    # set_sketch_params seed should override seed set by xkcd
    with plt.xkcd(3, 120, 40, 516):
        ln_3 = ax.plot(x, y_3, color='red')
        ln_3[0].set_sketch_params(3, 120, 40, 1234)
```
![artist_seed](https://github.com/matplotlib/matplotlib/assets/77437286/31556800-67ca-4877-893a-13e81a990936)


```
@image_comparison(['xkcd_seed.png'], remove_text=True)
def test_xkcd_seed():
    x = np.linspace(0.7, 1.42, 100)
    y_1 = x ** 2
    y_2 = x ** 3 + 1

    fig, ax = plt.subplots()

    with plt.xkcd(seed=2002):
        ln = ax.plot(x, y_1, color='black')
        ln[0].set_sketch_params(3, 120, 40)

    # xkcd should override rcParam seed.
    rcParams['path.sketch_seed'] = 0
    with plt.xkcd(3, 120, 40, seed=420):
        ln_2 = ax.plot(x, y_2, color='orange')
```
![xkcd_seed](https://github.com/matplotlib/matplotlib/assets/77437286/ad77e76a-f9be-413c-a3d4-8e0bd0441426)

```
@image_comparison(['path_seed.png'], remove_text=True)
def test_path_seed():
    x = x = np.linspace(0, 5, 200)
    y_1 = 20*np.sin(x)
    y_2 = 20*np.sin(x)+10
    fig, ax = plt.subplots()
    rcParams['path.sketch_seed'] = 645
    ln = ax.plot(x, y_1, color='blue')
    ln[0].set_sketch_params(3, 120, 40)
    rcParams['path.sketch_seed'] = 365
    ln = ax.plot(x, y_2, color='red')
    ln[0].set_sketch_params(3, 120, 40)
```
![path_seed](https://github.com/matplotlib/matplotlib/assets/77437286/de299d9f-a177-42d4-b489-bbb439d7c5a3)

First one tests seed updation through set_sketch_params
Second one tests seed updation through xkcd
Third one tests seed updation by directly changing the rcParam

Reply:
> rcParams['path.sketch_seed'] = 0 <this should be the default and possibly you may want to assert 

Have added a test to check the default value:
https://github.com/AryanSheka/matplotlib/blob/43947b7b0e65d7610c823855e67522943f2d6c81/lib/matplotlib/tests/test_path.py#L245

> RcParams[`path.sketch_seed`] == 0, but if you have to set it then that points to issues with the code.
The lines in the test (https://github.com/AryanSheka/matplotlib/blob/6a1d20e336ccd4c32d489cd7efbfe8ccabd7a559/lib/matplotlib/tests/test_path.py#L287)
was supposed to indicate setting the rcParam to an arbitrary number to test if it will get overridden by xkcd and set_sketch. I have changed it to arbitrary numbers and added comments which should make it clear.

Basically what I was trying to test is
```
rcParam['path.sketch_seed'] = arbitrary number
with plt.xkcd(seed= new_number)
```
should use new number as seed and not the value of rcParam. But if seed is not passed in xkcd or set_sketch, should use the value of rcParam as seed which by default is 0 unless changed by the user.

Reply:
That is already tested in this test:
https://github.com/AryanSheka/matplotlib/blob/43947b7b0e65d7610c823855e67522943f2d6c81/lib/matplotlib/tests/test_path.py#L250

The test in the previous comment is there to make sure that the seed passed through xkcd is used and not the arbitrary value.
I had added that test because while developing the code for this pr, I had encountered cases where the rcParam would get updated to the seed passed through xkcd, but it wouldn't be used as seed for drawing the figure ,so in the end if we checked value of rcParam, it would be new_number but seed used would be the initial arbitrary number. 

These were usually caused by mistakes in other parts of my code which I had to then rectify. 
Hence I believe for future changes, this is a good test to make sure everything is right .

Reply:
Also we will know that setting the seed to an arbitrary number by writing
```
rcParam['path.sketch_seed'] = arbitrary number
```
(and writing nothing else that will change the seed)

will change the figure and use the arbitrary number as seed  because if it didn't, then this test would fail:
https://github.com/AryanSheka/matplotlib/blob/43947b7b0e65d7610c823855e67522943f2d6c81/lib/matplotlib/tests/test_path.py#L326C1-L326C1